src/main/java/io/github/syst3ms/skriptparser/expressions/ExprContextValue.java

```diff
@@ -0,0 +1,73 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import io.github.syst3ms.skriptparser.Main;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import io.github.syst3ms.skriptparser.registration.ContextValueManager;
+import org.jetbrains.annotations.Nullable;
+
+import static io.github.syst3ms.skriptparser.registration.ContextValueManager.ContextValue;
+
+/**
+ * A specific context value.
+ * Refer to the documentation of the event to see which values can be used.
+ *
+ * @name Context Value
+ * @pattern [the] (past|future|) context-<.+>
+ * @since ALPHA
+ * @author Mwexim
+ */
+public class ExprContextValue implements Expression<Object> {
+
+	static {
+		Main.getMainRegistration().addExpression(
+			ExprContextValue.class,
+			Object.class,
+			false,
+			3,
+			"[the] (1:past|2:future|3:) context-<.+>"
+		);
+	}
+
+	private String name;
+	private int timeline;
+	private ContextValue<?> value;
+
+	@Override
+	public boolean init(Expression<?>[] vars, int matchedPattern, ParseContext parseContext) {
+		name = parseContext.getMatches().get(0).group();
+		if (parseContext.getParseMark() == 1) timeline = -1;
+		else if (parseContext.getParseMark() == 2) timeline = 1;
+		else timeline = 0;
+		for (Class<? extends TriggerContext> ctx : parseContext.getParserState().getCurrentContexts())
```

Comment:
This section of code doesn't comply with the rules I laid out in the previous comment either.